src/TestResult.dfy

```diff
@@ -49,4 +51,93 @@ module TestResult {
     var resourceCount :- Externs.ParseNat(resourceCountStr);
     Success(TestResult(displayName, outcome, durationTicks, resourceCount))
   }
-}
\ No newline at end of file
+
+  method GroupTestResults(results: seq<TestResult>) returns (groupedResults: map<string, seq<TestResult>>) {
+    groupedResults := map[];
+    for resIndex := 0 to |results| {
+      var res := results[resIndex];
+      var name := res.displayName;
+      if name in groupedResults {
+        groupedResults := groupedResults[name := groupedResults[name] + [res]];
+      } else {
+        groupedResults := groupedResults[name := [res]];
+      }
+    }
+  }
```

Comment:
Bonus points if you prove this equivalent to this:

```
method groupBy'<T(==)>(results: seq<T>, f: T -> string) returns (m: map<string, set<T>>) {
  var keys := set t: T | t in results :: f(t);
  m := map k: string | k in keys :: (set t: T | t in results && k == f(t));
}
```

Reply:
Maybe a nice thing to do soon! But in a future PR.